packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -161,6 +161,10 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
         };
         updateFocusContextKeys();
         this.shell.activeChanged.connect(updateFocusContextKeys);
+    }
+
+    async onStart(app: FrontendApplication): Promise<void> {
+        await this.workspacePreferences.ready;
```

Comment:
@federicobozzini should we instead opt for a non-blocking asynchronous call?

Example:

```ts

@inject(FrontendApplicationStateService)
protected readonly stateService: FrontendApplicationStateService;

async onStart(app: FrontendApplication): Promise<void> { 
    this.stateService.reachedState('ready').then(
         () => this.updateAddRemoveFolderActions(this.menuRegistry);
    );
}
```

Reply:
please don't use `FrontendApplicationStateService`, `FrontendApplicationContribution` provides already with sufficient lifecycle hooks

Reply:
@vince-fugnitto  I've made it non-blocking.

Reply:
@federicobozzini with this change I no longer see the toolbar menu item for `Add Folder to Workspace`, something which is added by:

```ts
this.updateAddRemoveFolderActions(this.menuRegistry);
```

In order for the item to return, one must right-click the empty area in the navigator for the command to exist. This is a regression from master and we may need to adjust.

Reply:
@vince-fugnitto Can you please be more specific about where that item should be? On master I see it only in the "Explorer" menu (just above the file navigator) and on the context menu opened on right-click in the navigator. I see the item in the same two places on this branch.

This is on the electron version.

Reply:
I performed additional testing and it does not look to be related to your changes.

---

For future reference, I was referencing the `...` (more toolbar item) from the Explorer widget which contains a menu item for `Add Folder to Workspace`:

<div align='center'>

![image](https://user-images.githubusercontent.com/40359487/71819855-43732c00-305b-11ea-947b-169d28fed5ef.png)

</div>